python/segyio/tools.py

```diff
@@ -402,3 +406,178 @@ def resample(f, rate = None, delay = None, micro = False,
     f._samples = (np.arange(len(f.samples)) * rate) + t0
 
     return f
+
+
+def create_from_array2D(filename, data):
+    """ Create a new SEGY file from a 2D array
+
+    Create an unstructured SEGY file from a 2-dimentional array with defaulted
+    headers. The tracecount and samplecount are inferred from the size of the
+    array. All traceheader fields are defaulted to zero. The file have a
+    defaulted textual header and binary header. The format is defaulted
+    to IBM FLOAT 4 BYTE, and iline, xline to 189 and 193, respectively.
+
+    The 2 array is interpreted as::
+
+                       samples
+                 --------------------
+        trace 0 | s0 | s1 | ... | sn |
+                 --------------------
+        trace 1 | s0 | s1 | ... | sn |
+                 --------------------
+                          .
+                          .
+                 --------------------
+        trace n | s0 | s1 | ... | sn |
+                 --------------------
+
+    traces = [0, len(axis(0)]
+    samples = [0, len(axis(1)]
+
+    Parameters
+    ----------
+
+    filename : str
```

Comment:
This actually only has to be string-like, i.e. we accept py.path, localpath and friends.